Source/Core/Core/HLE/HLE.cpp

```diff
@@ -189,6 +189,14 @@ u32 GetFunctionIndex(u32 addr)
   return (iter != s_original_instructions.end()) ? iter->second : 0;
 }
 
+u32 GetFirstFunctionIndex(u32 addr)
+{
+  u32 index = GetFunctionIndex(addr);
+  auto first = std::find_if(s_original_instructions.begin(), s_original_instructions.end(),
+                            [=](const auto& i) { return i.second == index && i.first < addr; });
```

Comment:
`i` would likely be better named something like `entry`, considering it's a map entry, not the instruction itself.